clang/lib/CIR/CodeGen/Address.h

```diff
@@ -104,16 +107,6 @@ class Address {
 
   bool hasOffset() const { return bool(offset); }
 
-  /// Return address with different element type, but same pointer and
-  /// alignment.
-  Address withElementType(mlir::Type ElemTy) const {
-    if (!hasOffset())
-      return Address(getBasePointer(), ElemTy, getAlignment(),
-                     getPointerAuthInfo(), /*Offset=*/nullptr,
-                     isKnownNonNull());
-    return Address(getPointer(), ElemTy, getAlignment(), isKnownNonNull());
```

Comment:
For parity with OG codegen perhaps with should keep the `withElementType` but wrap either `getBasePointer()` and `getPointer()` calls into creating a cast if the ElemTy != get*Pointer().getType().

Reply:
I'm not sure how to do that without giving the Address object a builder. Is there a way to create the bitcast here?

Reply:
It's fine if we have a `withElementType` takes a `CIRBaseBuilderTy` argument, a honest tradeoff for keeping the casts transparently while providing the same interface.

Reply:
OK. I'll rework the patch to do that.